File: python/tests/test_day1.py
from adoc23.day1 import solution1


def test_solution1_simple():
    all_input_str = """1abc2
pqr3stu8vwx
a1b2c3d4e5f
treb7uchet"""
    assert solution1(all_input_str) == 142


def test_solution1(shared_datadir):
    all_input = (shared_datadir / "day1.txt").read_text()
    assert solution1(all_input) == 54927

File: rust/tests/test_day1.rs
extern crate adoc2023;
use adoc2023::day1::{solution1, solution2};
use std::fs;
use std::path::PathBuf;

#[test]
fn test_solution1_simple() {
    let all_input = "1abc2\npqr3stu8vwx\na1b2c3d4e5f\ntreb7uchet";
    assert_eq!(solution1(all_input.to_string()), 142);
}

#[test]
fn test_solution1() {
    let file_path = PathBuf::from("tests/data/day1.txt");
    let all_input = fs::read_to_string(file_path).expect("Failed to read file");
    assert_eq!(solution1(all_input.to_string()), 54927);
}

#[test]
fn test_solution2_simple() {
    let all_input = "two1nine\neightwothree\nabcone2threexyz\nxtwone3four\n4nineeightseven2\nzoneight234\n7pqrstsixteen";
    assert_eq!(solution2(all_input.to_string()), 281);
}
